### PR TITLE
[mdatagen]: Use cobra for the command, add version flag

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: cmd/mdatagen/internal/templates/telemetry.go.tmpl [hunk 1]
```diff
@@ -14,7 +14,6 @@ import (
     {{- end }}
 
 	"go.opentelemetry.io/otel/metric"
-    "go.opentelemetry.io/otel/metric/noop"
```
Comment (atoulme) [inline cmd/mdatagen/internal/templates/telemetry.go.tmpl:17]: Should this be its own PR?

Comment (bogdandrutu) [reply cmd/mdatagen/internal/templates/telemetry.go.tmpl:17]: For you https://github.com/open-telemetry/opentelemetry-collector/pull/11197 :)

Comment (atoulme) [reply cmd/mdatagen/internal/templates/telemetry.go.tmpl:17]: appreciated!

File: cmd/mdatagen/README.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ You can run `cd cmd/mdatagen && $(GOCMD) install .` to install the `mdatagen` to
 
 ## Contributing to the Metadata Generator
 
-The code for generating the documentation can be found in [loader.go](./internal/loader.go) and the templates for rendering the documentation can be found in [templates](internal/templates).
+The code for generating the documentation can be found in [loader.go](./internal/loader.go) and the templates for rendering the documentation can be found in [templates](./internal/templates).
```
Comment (atoulme) [inline cmd/mdatagen/README.md:60]: thanks for fixing the link

**Review by atoulme — APPROVED** [review]
LGTM. Your move of the code from main.go to internal/command.go shows we don't have enough test coverage for some of the templating functions, something we can spin into a new issue.
